test/src/unit-request-handlers.cc

```diff
@@ -263,4 +377,52 @@ ArraySchema HandleLoadArraySchemaRequestFx::call_handler(
       stype, resp_buf->buffer());
 }
 
+shared_ptr<ArraySchema> HandleQueryPlanRequestFx::create_schema() {
+  auto schema = make_shared<ArraySchema>(HERE(), ArrayType::DENSE);
+  schema->set_capacity(10000);
+  throw_if_not_ok(schema->set_cell_order(Layout::ROW_MAJOR));
+  throw_if_not_ok(schema->set_tile_order(Layout::ROW_MAJOR));
+  uint32_t dim_domain[] = {1, 10, 1, 10};
+
+  auto dim1 = make_shared<Dimension>(HERE(), "dim1", Datatype::INT32);
+  throw_if_not_ok(dim1->set_domain(&dim_domain[0]));
+  auto dim2 = make_shared<Dimension>(HERE(), "dim2", Datatype::INT32);
+  throw_if_not_ok(dim2->set_domain(&dim_domain[2]));
+
+  auto dom = make_shared<Domain>(HERE());
+  throw_if_not_ok(dom->add_dimension(dim1));
+  throw_if_not_ok(dom->add_dimension(dim2));
+  throw_if_not_ok(schema->set_domain(dom));
+
+  auto attr1 = make_shared<Attribute>(HERE(), "attr1", Datatype::INT32);
+  throw_if_not_ok(schema->add_attribute(attr1));
+  auto attr2 = make_shared<Attribute>(HERE(), "attr2", Datatype::INT32);
+  throw_if_not_ok(schema->add_attribute(attr2));
+  auto attr3 = make_shared<Attribute>(HERE(), "attr3", Datatype::INT64);
+  throw_if_not_ok(schema->add_attribute(attr3));
+
+  return schema;
+}
+
+std::string HandleQueryPlanRequestFx::call_handler(
+    SerializationType stype, Query& query) {
+  auto ctx = tiledb::Context();
+  auto array = tiledb::Array(ctx, uri_.to_string(), TILEDB_READ);
+  auto req_buf = tiledb_buffer_handle_t::make_handle();
```

Comment:
do we need to `break_handle` afterwards?

Reply:
good catch, fixed, thanks!